### DISCUSSION TITLE
Not able to dynamically add options

### DISCUSSION THREAD

**Discussion by FarhadG** [pmndrs/leva#417] [Q&A] [answered]
I've noticed that I need to pre-define all of the options to be able to use the controls. I have a case where I need to dynamically add control values based on later user interactions. Some of these values may not be known ahead of time. Any tips on how to accomplish this?

Thank you for a great module! Looks great!

**Comment by dbismut** [comment, marked as answer]
If you pass a dependency array as the last argument to useControls, the input schema will be refreshed every time one of the dependencies in the array change.

**Reply by FarhadG** [reply]
Worked like a charm. Thank you!

**Reply by geyang** [reply]
The order fo the components gets shuffled.
